pylot/perception/depth_estimation/depth_est_operator.py

```diff
@@ -0,0 +1,146 @@
+        'lr':1e-3
+       }
+
+class DepthEstOperator(Op):
+    """ Subscribes to mid and right camera streams, and estimates depth using AnyNet."""
+    def __init__(self,
+                 name,
+                 output_stream_name,
+                 flags,
+                 log_file_name=None,
+                 csv_file_name=None):
+        super(DepthEstOperator, self).__init__(name)
+        self._flags = flags
+        self._left_camera_msgs = deque()
+        self._right_camera_msgs = deque()
+        self._left_imgs = deque()
+        self._right_imgs = deque()
+        self._logger = setup_logging(self.name, log_file_name)
+        self._csv_logger = setup_csv_logging(self.name + '-csv', csv_file_name)
+        self._output_stream_name = output_stream_name
+        self._lock = threading.Lock()
+        #load AnyNet
+        model = anynet.anynet.AnyNet(args)
+        model = nn.DataParallel(model).cuda()
+
+        if os.path.isfile(args['pretrained']):
+            checkpoint = torch.load(args['pretrained'])
+            model.load_state_dict(checkpoint['state_dict'])
+
+        if os.path.isfile(args['resume']):
+            checkpoint = torch.load(args['resume'])
+            model.load_state_dict(checkpoint['state_dict'])
+            
+        self._model = model
+
+    @staticmethod
+    def setup_streams(input_streams, output_stream_name,
+                      left_camera_name, right_camera_name):
+        # Select camera input streams.
+        camera_streams = input_streams.filter(is_camera_stream)
+        camera_streams.filter_name(left_camera_name).add_callback(
+            DepthEstOperator.on_left_camera_msg)
+        camera_streams.filter_name(right_camera_name).add_callback(
+            DepthEstOperator.on_right_camera_msg)
+        input_streams.add_completion_callback(
+            DepthEstOperator.on_watermark)
+        return [create_depth_est_stream(output_stream_name)]
+
+    def on_left_camera_msg(self, msg):
+        with self._lock:
+            self._left_camera_msgs.append(msg)
+            img = rgb_to_bgr(msg.frame).astype(np.uint8) # used as bgr_to_rgb, need RGB frames
+            img = img[200:500] # crop
+            processed = preprocess.get_transform(augment=False)  
+            img = processed(img)
+            self._left_imgs.append(img)
+            if self._right_imgs:
+                self.eval_depth(msg)
+        
+    def on_right_camera_msg(self, msg):
+        with self._lock:
+            self._right_camera_msgs.append(msg)
+            img = rgb_to_bgr(msg.frame).astype(np.uint8) # used as bgr_to_rgb, need RGB frames
+            img = img[200:500] # crop
+            processed = preprocess.get_transform(augment=False)  
+            img = processed(img)
+            self._right_imgs.append(img)
+            if self._left_imgs:
+                self.eval_depth(msg)
+        
+    def on_watermark(self, msg):
```

Comment:
I see you're not doing anything on_watermarks. You don't need to implement this method, and thus you also don't need self._left_camera_msgs and self._right_camera_msgs.